plugins/org.jkiss.dbeaver.ext.timeplus/src/org/jkiss/dbeaver/ext/timeplus/TimeplusTypeParser.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.timeplus;
+
+import com.google.gson.Gson;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.jkiss.code.NotNull;
+import org.jkiss.code.Nullable;
+import org.jkiss.dbeaver.DBException;
+import org.jkiss.dbeaver.Log;
+import org.jkiss.dbeaver.ext.timeplus.TimeplusDataTypesParser.ArrayTypeContext;
+import org.jkiss.dbeaver.ext.timeplus.TimeplusDataTypesParser.MapTypeContext;
+import org.jkiss.dbeaver.ext.timeplus.TimeplusDataTypesParser.TupleTypeContext;
+import org.jkiss.dbeaver.ext.timeplus.model.TimeplusArrayType;
+import org.jkiss.dbeaver.ext.timeplus.model.TimeplusDataSource;
+import org.jkiss.dbeaver.ext.timeplus.model.TimeplusMapType;
+import org.jkiss.dbeaver.ext.timeplus.model.TimeplusTupleType;
+import org.jkiss.dbeaver.ext.timeplus.model.data.TimeplusMapValue;
+import org.jkiss.dbeaver.ext.timeplus.model.data.TimeplusTupleValue;
+import org.jkiss.dbeaver.model.DBUtils;
+import org.jkiss.dbeaver.model.data.json.JSONUtils;
+import org.jkiss.dbeaver.model.exec.DBCSession;
+import org.jkiss.dbeaver.model.runtime.DBRProgressMonitor;
+import org.jkiss.dbeaver.model.struct.DBSDataType;
+import org.jkiss.dbeaver.model.struct.DBSTypedObject;
+import org.jkiss.utils.CommonUtils;
+import org.jkiss.utils.Pair;
+
+import java.io.StringReader;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class TimeplusTypeParser {
```

Comment:
This and the following parts of code look exactly like for Clickhouse. Please avoid copy-paste and duplication. You can inherit from clickhouse classes.